src/com/SocketSendQueue.hpp

```diff
@@ -36,9 +37,12 @@ class SocketSendQueue {
     std::function<void()>        callback;
   };
 
+  /// The queue, containing items to asynchronously send using boost.asio.
   std::deque<SendItem> _itemQueue;
-  std::mutex           _sendMutex;
-  bool                 _ready = true;
+  /// The mutex protecting access to the queue
+  std::mutex _queueMutex;
+  /// Is the queue allowed to start another asynchronous send?
+  std::atomic_bool _ready{true};
```

Comment:
Do we need the atomic boolean here, considering that the queue is now locked anyway?

Reply:
Good point. This is due to strange interface, I'll clarify it.